src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -71,20 +72,31 @@ public sealed partial class TextLoader
             char separator = '\t', bool allowQuotedStrings = true,
             bool supportSparse = true, bool trimWhitespace = false)
         {
-            var fields = typeof(TInput).GetFields();
-            Arguments.Column = new TextLoaderColumn[fields.Length];
-            for (int index = 0; index < fields.Length; index++)
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields();
+
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.CanWrite && x.GetIndexParameters().Length == 0)
+                .ToArray();
+
+            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+
+            Arguments.Column = new TextLoaderColumn[memberInfos.Length];
+            for (int index = 0; index < memberInfos.Length; index++)
             {
-                var field = fields[index];
-                var mappingAttr = field.GetCustomAttribute<ColumnAttribute>();
+                var memberInfo = memberInfos[index];
+                var mappingAttr = memberInfo.GetCustomAttribute<ColumnAttribute>();
                 if (mappingAttr == null)
-                    throw Contracts.Except($"{field.Name} is missing ColumnAttribute");
+                    throw Contracts.Except($"field or property {memberInfo.Name} is missing ColumnAttribute");
```

Comment:
>f [](start = 45, length = 1)

It's a totally nitpicking, but we prefer to start our sentences with Capital letter.